### PR TITLE
add new methods to Reflection

### PR DESCRIPTION
**Pull request by alex8b**
fix IsSpectating
fix IsFriendsListVisible
add IsGameMenuVisible
add IsOptionsMenuVisible
add IsMulligan
add NumMulliganCards
add IsChoosingCard
add NumChoiceCards
add IsPlayerEmotesVisible
add IsEnemyEmotesVisible
add IsInBattlecryEffect
add IsDragging
add IsTargetingHeroPower
add BattlecrySourceCardZonePosition
add IsHoldingCard
add IsTargetReticleActive
add IsEnemyTargeting
add IsGameOver
add IsInMainMenu
add GetShownUiWindowId
add IsPlayerHandZoneUpdatingLayout
add IsPlayerPlayZoneUpdatingLayout
add GetCurrentSceneMode
add GetNumCardsPlayerHand
add GetNumCardsPlayerBoard
add GetNavigationHistorySize

### CODE REVIEW

File: HearthMirror/Reflection.cs [hunk 1]
```diff
@@ -230,7 +230,56 @@ private static Deck GetEditedDeckInternal()
...
+		public static bool IsOptionsMenuVisible() => TryGetInternal(() => Mirror.Root["OptionsMenu"]?["s_instance"]?["m_isShown"]) ?? false;
+
+		public static bool IsMulligan() => TryGetInternal(() => Mirror.Root["MulliganManager"]?["s_instance"]?["mulliganChooseBanner"]) != null;
+
+		public static int GetNumMulliganCards() => TryGetInternal(() => Mirror.Root["MulliganManager"]?["s_instance"]?["m_startingCards"]?["_size"]) ?? 0;
+
+		public static bool IsChoosingCard() => (TryGetInternal(() => Mirror.Root["ChoiceCardMgr"]?["s_instance"]?["m_subOptionState"]) != null)
+				|| ((int)(TryGetInternal(() => Mirror.Root["ChoiceCardMgr"]?["s_instance"]?["m_choiceStateMap"]?["count"]) ?? 0) > 0);
+
+		public static int GetNumChoiceCards() => TryGetInternal(() => Mirror.Root["ChoiceCardMgr"]?["s_instance"]?["m_lastShownChoices"]?["_size"]) ?? 0;
```
Comment (azeier) [inline HearthMirror/Reflection.cs]: Why do you use `m_lastShownChoices` over `m_lastShownChoices` as in `IsChoosingCard()` here?

Also, why not use this `>0` over the 2 checks in `IsChoosingCard()`?

Comment (alex8b) [inline HearthMirror/Reflection.cs]: I think they don't clean it up when they close the overlay so `m_lastShownChoices` is not zero when overlay is not visible. Is it assigned only in `ChoiceCardMgr.ShowChoices()`

The first one is basically copied from `ChoiceCardMgr.IsShown()`

Comment (azeier) [inline HearthMirror/Reflection.cs]: Then why not use `m_choiceStateMap.count` for NumChoices as well? This just seems a bit inconsistent either way :).

Comment (alex8b) [inline HearthMirror/Reflection.cs]: `m_choiceStateMap.count` is a number of choice states, not the number of cards shown on the screen. For example it's used in _Choose Discover Spell or Discover Minion_ spell, that has two sub states, one with 2 cards and another one with 3 cards.

```
  private class ChoiceState
  {
    public List<Card> m_cards = new List<Card>();
    public bool m_waitingToShow;
    public PowerTaskList m_preTaskList;
  }
```

Comment (azeier) [inline HearthMirror/Reflection.cs]: Oh I see., thanks for clarifying.

Comment (alex8b) [inline HearthMirror/Reflection.cs]: Great! Thanks for review and merging!

**Pull request merged**